### PR TITLE
Namespace fixes.

### PR DESCRIPTION
**Pull request by tmihalik**

### CODE REVIEW

File: classes/config.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static function load($file, $group = null, $reload = false, $overwrite =
 			}
 		}
 
-		if ($file instanceof Config_Interface)
+		if ($file instanceof \Config_Interface)
```
Comment (jschreuder) [inline classes/config.php]: This is wrong, `Config_Interface` in global should extend `Fuel\Core\Config_Interface` thus the current way is more inclusive.

File: classes/database/connection.php [hunk 2]
```diff
@@ -439,12 +439,12 @@ public function quote($value)
 		}
 		elseif (is_object($value))
 		{
-			if ($value instanceof Database_Query)
+			if ($value instanceof \Database_Query)
```
Comment (jschreuder) [inline classes/database/connection.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/database/connection.php [hunk 3]
```diff
@@ -439,12 +439,12 @@ public function quote($value)
...
 			{
 				// Create a sub-query
 				return '('.$value->compile($this).')';
 			}
-			elseif ($value instanceof Database_Expression)
+			elseif ($value instanceof \Database_Expression)
```
Comment (jschreuder) [inline classes/database/connection.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/database/connection.php [hunk 4]
```diff
@@ -498,7 +498,7 @@ public function quote_table($value)
 		}
 
 		// deal with the sub-query objects first
-		if ($table instanceof Database_Query)
+		if ($table instanceof \Database_Query)
```
Comment (jschreuder) [inline classes/database/connection.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/database/connection.php [hunk 5]
```diff
@@ -580,12 +580,12 @@ public function quote_identifier($value)
 		}
 		elseif (is_object($value))
 		{
-			if ($value instanceof Database_Query)
+			if ($value instanceof \Database_Query)
```
Comment (jschreuder) [inline classes/database/connection.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/database/connection.php [hunk 6]
```diff
@@ -580,12 +580,12 @@ public function quote_identifier($value)
...
 			{
 				// Create a sub-query
 				return '('.$value->compile($this).')';
 			}
-			elseif ($value instanceof Database_Expression)
+			elseif ($value instanceof \Database_Expression)
```
Comment (jschreuder) [inline classes/database/connection.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/database/query.php [hunk 7]
```diff
@@ -240,7 +240,7 @@ public function execute($db = null)
 				$result = $cache->get();
 				return new Database_Result_Cached($result, $sql, $this->_as_object);
 			}
-			catch (CacheNotFoundException $e) {}
+			catch (\CacheNotFoundException $e) {}
```
Comment (jschreuder) [inline classes/database/query.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/date.php [hunk 8]
```diff
@@ -138,8 +138,8 @@ public static function create_from_string($input, $pattern_key = 'local')
 	 */
 	public static function range_to_array($start, $end, $interval = '+1 Day')
 	{
-		$start     = ( ! $start instanceof Date) ? static::forge($start) : $start;
-		$end       = ( ! $end instanceof Date) ? static::forge($end) : $end;
+		$start     = ( ! $start instanceof \Date) ? static::forge($start) : $start;
```
Comment (jschreuder) [inline classes/date.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/date.php [hunk 9]
```diff
@@ -138,8 +138,8 @@ public static function create_from_string($input, $pattern_key = 'local')
...
+		$end       = ( ! $end instanceof \Date) ? static::forge($end) : $end;
```
Comment (jschreuder) [inline classes/date.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/fieldset.php [hunk 10]
```diff
@@ -162,7 +162,7 @@ protected function __construct($name, array $config = array())
 	 */
 	public function validation($instance = true)
 	{
-		if ($instance instanceof Validation)
+		if ($instance instanceof \Validation)
```
Comment (jschreuder) [inline classes/fieldset.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/fieldset.php [hunk 11]
```diff
@@ -184,7 +184,7 @@ public function validation($instance = true)
 	 */
 	public function form($instance = true)
 	{
-		if ($instance instanceof Form)
+		if ($instance instanceof \Form)
```
Comment (jschreuder) [inline classes/fieldset.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/fieldset.php [hunk 12]
```diff
@@ -254,7 +254,7 @@ protected function add_child(Fieldset $fieldset)
 	 */
 	public function add($name, $label = '', array $attributes = array(), array $rules = array())
 	{
-		if ($name instanceof Fieldset_Field)
+		if ($name instanceof \Fieldset_Field)
```
Comment (jschreuder) [inline classes/fieldset.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/fieldset.php [hunk 13]
```diff
@@ -265,7 +265,7 @@ public function add($name, $label = '', array $attributes = array(), array $rule
 			$this->fields[$name->name] = $name;
 			return $name;
 		}
-		elseif ($name instanceof Fieldset)
+		elseif ($name instanceof \Fieldset)
```
Comment (jschreuder) [inline classes/fieldset.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/fieldset/field.php [hunk 14]
```diff
@@ -81,7 +81,7 @@ class Fieldset_Field
 	public function __construct($name, $label = '', array $attributes = array(), array $rules = array(), $fieldset = null)
 	{
 		$this->name = (string) $name;
-		$this->fieldset = $fieldset instanceof Fieldset ? $fieldset : null;
+		$this->fieldset = $fieldset instanceof \Fieldset ? $fieldset : null;
```
Comment (jschreuder) [inline classes/fieldset/field.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

File: classes/file.php [hunk 15]
```diff
@@ -70,7 +70,7 @@ public static function forge(array $config = array())
 	 */
 	public static function instance($area = null)
 	{
-		if ($area instanceof File_Area)
+		if ($area instanceof \File_Area)
```
Comment (jschreuder) [inline classes/file.php]: Same as earlier comment, instanceof should use the most basic form not the extended.

**Comment by jschreuder** [comment]
Ok, I won't mark them all but you get the point.

**Comment by tmihalik** [comment]
What about these? 
https://github.com/fuel/core/blob/1.1/develop/classes/database/query.php#L182
https://github.com/fuel/core/blob/1.1/develop/classes/database/query/builder/select.php#L294
https://github.com/fuel/core/blob/1.1/develop/classes/request.php#L476
https://github.com/fuel/core/blob/1.1/develop/classes/controller/template.php#L53
etc...

**Comment by jschreuder** [comment]
Same for each typehint & instanceof, those should always use the Fuel\Core classes as those are the base and will be the most inclusive.

**Comment by WanWizard** [comment]
No response on this topic. Request can no longer be merged.

**Comment by jschreuder** [comment]
There's a few in here that are still relevant, let's keep it open as a reminder.

**Comment by frankdejonge** [comment]
Fixed these namespace issues, closing this pull request.

**Pull request closed without merging**